lib/index.js

```diff
@@ -179,6 +200,39 @@ FSTree.defaultIsEqual = function defaultIsEqual(entryA, entryB) {
   return equal;
 };
 
+FSTree.applyPatch = function(input, output, patch, _delegate) {
+  var delegate = assign({}, DEFAULT_DELEGATE, _delegate);
+  for (var i = 0; i < patch.length; i++) {
+    applyOperation(input, output, patch[i], delegate);
+  }
+};
+
+function applyOperation(input, output, operation, delegate) {
+  var method = operation[0];
+  var relativePath = operation[1];
+  var inputPath = path.join(input, relativePath);
+  var outputPath = path.join(output, relativePath);
+
+  if (delegate[method]) {
+    delegate[method](inputPath, outputPath, relativePath);
+  } else {
+    throw new Error('Unknown patch operation method: ' + method);
+  }
+}
+
+function copy(sourcePath, outputPath) {
+  var outputDir = path.dirname(outputPath);
+  try {
+    symlinkOrCopy.sync(sourcePath, outputPath);
+  } catch (e) {
+    try {
```

Comment:
can we please add some comments to this code block to explain why we're being so defensive here, both

- why we expect `symLinkOrCopy` to fail during normal operation and
- why we completely swallow exceptions from `fs.unlinkSync`